### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by IstiakAhmed-1721365**
Getting "Not found. Authentication passthru." error while trying to authenticate. Please check and give some feedback, thank you.

### CODE REVIEW

File: app/controllers/users/unlocks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Users
+  class UnlocksController < Devise::UnlocksController
+    # GET /resource/unlock/new
```
Comment (CW-AnhNH) [inline app/controllers/users/unlocks_controller.rb:5]: remove unnecessary comments and commented code

File: app/controllers/users/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Users
+  class SessionsController < Devise::SessionsController
```
Comment (CW-AnhNH) [inline app/controllers/users/sessions_controller.rb:4]: remove unnecessary comments and commented code

File: app/controllers/users/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Users
+  class RegistrationsController < Devise::RegistrationsController
```
Comment (CW-AnhNH) [inline app/controllers/users/registrations_controller.rb:4]: remove unnecessary comments and commented code

File: Gemfile [hunk 4]
```diff
@@ -75,3 +76,7 @@ group :test do
 end
 
 gem "devise", "~> 4.8"
+
+gem "omniauth"
+gem 'omniauth-google-oauth2'
+gem "omniauth-rails_csrf_protection"
```
Comment (CW-AnhNH) [inline Gemfile:82]: should end with a newline

File: Gemfile [hunk 5]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (CW-AnhNH) [inline Gemfile:1]: remove empty line

File: app/models/comment.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 class Comment < ApplicationRecord
-    belongs_to : user, post, group
-    has_many : active_relationship, foreign_key: "user_id", "post_id"
+  validates :content, presence :true
```
Comment (CW-AnhNH) [inline app/models/comment.rb:2]: syntax error
`presence :true ->> presence: true`

File: app/controllers/users/sessions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_sign_in_params
+    #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
+    # end
+    def after_sign_out_path_for(_resource_or_scope)
+      new_user_session_path
+    end
+
+    def after_sign_in_path_for(resource_or_scope)
+      stored_location_for(resource_or_scope) || root_path
```
Comment (CW-AnhNH) [inline app/controllers/users/sessions_controller.rb:33]: Have you tested that function?

File: db/migrate/20220612170820_add_columns_to_users.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class AddColumnsToUsers < ActiveRecord::Migration[7.0]
+  def change
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply db/migrate/20220612170820_add_columns_to_users.rb:2]: tried this while implementing google_oauth but it didn't work so I commented it out.

**Pull request closed without merging**